packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -281,6 +296,16 @@ - (NSView *)nextViewToFocusOutsideZone:(FocusZoneAction)action
 	{
 		nextViewToFocus = nil;
 	}
+	else
+	{
+		// If the next view is in a FocusZone, focus the default tabbable element.
+		RCTFocusZone *nextFocusZone = GetFocusZoneAncestor(nextViewToFocus);
+		NSView *element = [nextFocusZone defaultTabbableElement];
+		if (element != nil)
+		{
+			nextViewToFocus = element;
+		}
```

Comment:
Note: This is speculation (and I also know very little about the responder chain), so take it with many grains of salt, and don't consider it blocking. 

I wonder if the more "macOS" way to do this would have been to remember who the last responder was inside an override of "acceptsFirstResponder" / "resignFirstResponder" or something else to do with overriding the responder chain. I think if we did it that way, we would only have the reference to the last focused element in native and not in JS so it wouldn't work.